File: 551.StudentAttendanceRecordI/record.py
class Solution(object):
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        countA = 0
        countL = 0
        for c in s:
            if c == 'L':
                countL += 1
            else:
                countL = 0
                if c == 'A':
                    countA += 1

            if countA > 1 or countL > 2:
                return False
        return True

File: 551.StudentAttendanceRecordI/record.cc
class Solution {
public:
    bool checkRecord(string s) {
        int countA = 0, countL = 0;
        for (char c : s) {
            if (c == 'L') {
                countL++;
            }
            else {
                countL = 0;
                if (c == 'A') {
                    countA++;
                }
            }

            if (countA > 1 || countL > 2) {
                return false;
            }
        }

        return true;
    }
};
